pipettes/firmware/interfaces.cpp

```diff
@@ -6,9 +6,83 @@
 #include "platform_specific_hal_conf.h"
 #pragma GCC diagnostic pop
 
-auto interfaces::driver_config_by_axis(PipetteType which)
-    -> tmc2130::configs::TMC2130DriverConfig {
+auto interfaces::driver_config_by_axis(PipetteAxisType which)
+    -> driver_utils::DriverConfigType {
     switch (which) {
+        case PipetteAxisType::left_gear_motor:
+            return tmc2130::configs::TMC2130DriverConfig{
+                .registers = {.gconfig = {.en_pwm_mode = 1},
```

Comment:
These configurations look really similar. Is there a way to compose these structs in a way that reduces some of the repetition?